.github/workflows/e2e_ci_workflow.yml

```diff
@@ -0,0 +1,94 @@
+# This workflow will install Python dependencies and run end to end tests with single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: CI E2E tests
+
+on:
+  schedule:
+    # Monday to Thursday 1 AM PDT build
+    # * is a special character in YAML so you have to quote this string
+    - cron: "0 8 * * 1,2,3,4"
+  workflow_dispatch:
+  push:
+    branches: [ release/* ]
+  pull_request:
+    branches: [ dev, master, main ]
+
+jobs:
+  build:
+    name: "Python E2E CI Run"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: 
+        python-version: [ 3.6, 3.7, 3.8, 3.9 ]
+
+    steps:
+      - name: Checkout code.
+        uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Set up Dotnet 2.2.207
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '2.2.207'
+      - name: Install dependencies and the worker
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple -U -e .[dev]
+          python -m pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple -U azure-functions --pre
+          python setup.py build
+          python setup.py webhost
+      - name: Running 3.6 Tests
+        if: matrix.python-version == 3.6
```

Comment:
GHA doesn't have a good way to have selective environment variables, like the way we had in AzDO. This was the second-best approach I could come up with, after multiple attempts. I didn't want to have multiple workflow files for the E2E.